### PR TITLE
fix replicate when pk/unique index is a generated column

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 

Fix https://github.com/pingcap/ticdc/issues/354

### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change

Side effects

 - 

Related changes

 - No

### Release note

- cdc: fix replicate when pk/unique index is a generated column

### CODE REVIEW

File: tests/generate_column/data/prepare.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+drop database if exists `generate_column`;
+create database `generate_column`;
+use `generate_column`;
+
+create table t (a int, b int as (a + 1) stored primary key);
+insert into t(a) values (1),(2), (3);
+update t set a = 10 where a = 1;
+
```
Comment (leoppro) [inline tests/generate_column/data/prepare.sql:8]: add a delete sql here

File: tests/generate_column/run.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        mysql) ;&
+        *) SINK_URI="mysql://root@127.0.0.1:3306/";;
+    esac
+    cdc cli changefeed create --start-ts=$start_ts --sink-uri="$SINK_URI"
+    if [ "$SINK_TYPE" == "kafka" ]; then
+      run_kafka_consumer $WORK_DIR "kafka://127.0.0.1:9092/$TOPIC_NAME?partition-num=4"
+    fi
+    run_sql_file $CUR/data/prepare.sql ${UP_TIDB_HOST} ${UP_TIDB_PORT}
+    # sync_diff can't check non-exist table, so we check expected tables are created in downstream first
+    check_table_exists drop_tables.c ${DOWN_TIDB_HOST} ${DOWN_TIDB_PORT}
```
Comment (amyangfei) [inline tests/generate_column/run.sh:35]:
```suggestion
    check_table_exists generate_column.t ${DOWN_TIDB_HOST} ${DOWN_TIDB_PORT}
```

File: tests/generate_column/data/prepare.sql (hunk 1 above)
Comment (crazycs520) [reply tests/generate_column/data/prepare.sql:11]: done.

**Comment by amyangfei** [comment]
The kafka test case because the default kafka dispatcher panic here https://github.com/pingcap/ticdc/blob/master/cdc/sink/dispatcher/default.go#L31
row.Columns[row.IndieMarkCol]
`&{Ts:416614286032896005 RowID:5 Schema:generate_column Table:t Delete:false IndieMarkCol:b Columns:map[a:0xc0004bafc0] Keys:[`generate_column`.`t`]}`

The column fields don't include `b` but the dispatcher uses it as `IndieMarkCol`

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2cf79a0]

goroutine 578 [running]:
github.com/pingcap/ticdc/cdc/sink/dispatcher.(*defaultDispatcher).Dispatch(0xc000ec70ac, 0xc000381490, 0xc000000001)
        /home/tidb/code/ticdc/cdc/sink/dispatcher/default.go:31 +0x750
github.com/pingcap/ticdc/cdc/sink/dispatcher.(*dispatcherSwitcher).Dispatch(0xc000357520, 0xc000381490, 0xc000000001)
        /home/tidb/code/ticdc/cdc/sink/dispatcher/interface.go:59 +0x273
github.com/pingcap/ticdc/cdc/sink.(*mqSink).EmitRowChangedEvents(0xc00096e7e0, 0x432a2a0, 0xc0004fa4c0, 0xc0001c2ce0, 0x1, 0x1, 0x0, 0x0)
        /home/tidb/code/ticdc/cdc/sink/mq.go:87 +0x356
github.com/pingcap/ticdc/cdc.(*processor).syncResolved(0xc0000ec000, 0x432a2a0, 0xc0004fa4c0, 0x0, 0x0)
        /home/tidb/code/ticdc/cdc/processor.go:645 +0x74a
github.com/pingcap/ticdc/cdc.(*processor).Run.func4(0xc0007d2f68, 0x0)
        /home/tidb/code/ticdc/cdc/processor.go:217 +0x8e
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0007e4630, 0xc0007c27a0)
        /home/tidb/.gvm/pkgsets/go1.13.6/global/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57 +0x72
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/tidb/.gvm/pkgsets/go1.13.6/global/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:54 +0x74
```

**Comment by amyangfei** [comment]
The test case failure case is fixed in https://github.com/pingcap/ticdc/pull/566

**Pull request merged**